### PR TITLE
Inventory support for labels, server groups, server metadata and public/private IP addresses.

### PR DESCRIPTION
**Pull request by ajmyyra**
This PR brings in following updates:
* Servers without matching interface (`connect_with` that is `public_ipv4` by default) are skipped so other servers can be accessed. Skipped ones are logged.
* Support for filtering with server labels
* Support for filtering with server group

### CODE REVIEW

File: plugins/inventory/upcloud.py [hunk 1]
```diff
@@ -173,7 +211,28 @@ def _filter_servers(self):
...
+                disqualified = False
+                for wanted_label in self.get_option("labels"):
+                    server_labels = _parse_server_labels(server.labels['label'])
+                    if len(server_labels) == 0:
+                        disqualified = True
+
+                    for server_label in server_labels:
+                        display.vvvv(f"Comparing {wanted_label} against {server_label}")
+                        if wanted_label not in server_label:
+                            disqualified = True
```
Comment (Darep) [inline plugins/inventory/upcloud.py:227]: If server has e.g. labels: `['foo=foo', 'bar']` and we wanted label `['bar']`, we will disqualify the server as the first hit to check `'bar' in 'foo=foo'` will fail and set disqualified to True? 🤔

Comment (ajmyyra) [reply plugins/inventory/upcloud.py:227]: Current labels implementation checks for the existance of all labels, not if a server has some of them. Do you think it makes sense or should we change it to similar as zones where you specify multiple ones and they're all included?

Comment (Darep) [reply plugins/inventory/upcloud.py:227]: Not sure about the ansible-inventory mind model, but – to me – servers not showing up when i've put in at least 1 of the labels would be confusing 😄

Comment (ajmyyra) [reply plugins/inventory/upcloud.py]: Now changed to that. It's different from how tags work there, but as we're moving away from them anyway, it's not that big of an issue.

**Review by Darep — APPROVED** [review]
Did not test the code & I'm no expert with ansible-inventory, but changes LGTM!

**Pull request merged**